.circleci/config.yml

```diff
@@ -93,4 +95,36 @@ jobs:
             if [ $? -ne 0 ]; then
               echo "Unformatted code detected. Please run black --fast . locally and commit to format your code." >&2
               exit 1
-            fi
\ No newline at end of file
+            fi
+
+  ## ------------------ Mypy for Static Type Checking python code ------------------
+  mypy-type-checks:
+    parameters:
+      with_merge: # Part of the PR-merge workflow
+        type: boolean
+        default: false
+    docker:
+      # https://circleci.com/developer/images/image/cimg/python
+      - image: cimg/python:3.11.0
+    steps:
+      - when:
+          condition: << parameters.with_merge >>
+          steps:
+            - disable_build_if_not_pr
+      - checkout
+      # Run Pyre type checking
+      - run:
+          name: Run Static type checking
+          command: |
```

Comment:
This build is missing steps to store the resulting HTML report and xunit report.
```yaml
      - store_test_results:
          path: /tmp/workspace/junit/

      - store_artifacts:
          name: Save Unit Test Results
          path: /tmp/workspace/junit/
```

Reply:
Store the MyPy type checking results and artifacts.